code-challenges.test.js

```diff
@@ -10,42 +11,197 @@
 // Reminder: The test will call your function
 // Run the file with the following command: $ yarn jest
 
+
 // --------------------1) Create a function that takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
 
 // a) Create a test with an expect statement using the variable provided.
 
-const people = [
+
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("jobs", () => {
+
+  // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+  it(" that takes in an array of objects and returns an array with a sentence about each person with their name capitalized.", () => {
+
+
+ const people = [
   { name: "ford prefect", occupation: "a hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "a radio employee" }
 ]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
+    //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+    expect(jobs(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
+// function jobs takes array 
+// let results equal empty array
+// for index let equal zero 
+// index is less than length of array
+// itterate forward left to right
+// let newarray equal array index name split string
+// for let index equal 0 index is less than new arr length itterate to right
+// new array index equals 
+// string newarray join string is a array index occupation
+// return results push result
+
+ const jobs = (array) => {
+   let results = []
+    for (let i = 0; i < array.length; i++){
+      let newArray = array[i].name.split(" ")
+      for (let i = 0; i < newArray.length; i++){
+      newArray[i] = newArray[i].charAt(0).toUpperCase() + newArray[i].slice(1);
+        
+      }    
+      let result = `${newArray.join(" ")} is a ${array[i].occupation}.`
+    }
+    return results.push(result)
```

Comment:
Yeah I think the problem is on this line 61. If you return the push action you will get the length of the array. It is a weird side effect of the mutator method. If you push the results then return the results array separately, it should get you what you need.